internal/controller/controllers/hypershiftagentserviceconfig_controller_test.go

```diff
@@ -0,0 +1,83 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	"github.com/openshift/assisted-service/internal/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	kubeconfigSecret = "secret"
+)
+
+func newHypershiftAgentServiceConfigRequest(asc *aiv1beta1.HypershiftAgentServiceConfig) ctrl.Request {
+	namespacedName := types.NamespacedName{
+		Namespace: asc.ObjectMeta.Namespace,
+		Name:      asc.ObjectMeta.Name,
+	}
+	return ctrl.Request{NamespacedName: namespacedName}
+}
+
+func newHSCDefault() *aiv1beta1.HypershiftAgentServiceConfig {
+	baseAsc := newASCDefault()
+	return &aiv1beta1.HypershiftAgentServiceConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testName,
+			Namespace: testNamespace,
+		},
+		Spec: aiv1beta1.HypershiftAgentServiceConfigSpec{
+			AgentServiceConfigSpec: aiv1beta1.AgentServiceConfigSpec{
+				FileSystemStorage: baseAsc.Spec.FileSystemStorage,
+				DatabaseStorage:   baseAsc.Spec.DatabaseStorage,
+				ImageStorage:      baseAsc.Spec.ImageStorage,
+			},
+
+			RemoteKubeconfigRef: corev1.LocalObjectReference{
+				Name: kubeconfigSecret,
+			},
+		},
+	}
+}
+
+var _ = Describe("HypershiftAgentServiceConfig reconcile", func() {
+	var (
+		ctx      = context.Background()
+		c        client.Client
+		hr       *HypershiftAgentServiceConfigReconciler
+		mockCtrl *gomock.Controller
+	)
+	BeforeEach(func() {
+		c = fakeclient.NewClientBuilder().WithScheme(GetKubeClientSchemes()).Build()
+		mockCtrl = gomock.NewController(GinkgoT())
+
+		hr = &HypershiftAgentServiceConfigReconciler{
+			Client:    c,
+			Scheme:    c.Scheme(),
+			Log:       common.GetTestLog(),
+			Namespace: testNamespace,
+		}
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+	})
+
+	Context("Success", func() {
+		It("place holder test", func() {
+			hsc := newHSCDefault()
+			res, err := hr.Reconcile(ctx, newHypershiftAgentServiceConfigRequest(hsc))
+			Expect(err).To(BeNil())
+			Expect(res).To(Equal(ctrl.Result{}))
+		})
+	})
```

Comment:
This context is pointless.
```suggestion
	It("runs without error", func() {
		hsc := newHSCDefault()
		res, err := hr.Reconcile(ctx, newHypershiftAgentServiceConfigRequest(hsc))
		Expect(err).To(BeNil())
		Expect(res).To(Equal(ctrl.Result{}))
	})
```

Reply:
The context will be filled pretty soon with additional tests as in https://github.com/openshift/assisted-service/blob/f7ab8b3d4ff7f12434c85624fca1ba3553a35aae/internal/controller/controllers/agentserviceconfig_controller_test.go#L115. So I would really like to retain this place holer

Reply:
Done.